### PR TITLE
Added script to update an svg's id

### PR DESCRIPTION
**Pull request by Thomas-Boi**
Hey guys,

I'm working on a script to update the ids within an `svg`. This is to address #573 and will eventually be integrated as its own build task.

For now, I just want to get everyone's opinion on what I have. Try running the script as stated below and let me know what you think. 

Steps:
- Check the `/icons/elixir/elixir-test.svg` and `/icons/ruby/ruby-test.svg`. These are copied over from `/icons/elixir/elixir-original.svg` and `/icons/ruby/ruby-original.svg`.
  - You should search for "id" and "url(" to see the ids being used in these files. Notice that they are similar values like "a", "b", "c"...
- Run `python ./.github/scripts/update_id.py`. This will convert the ids within this file to the format `filename-num` where `num` is an incrementing number. 
  - You can also check the `update_id.py` and ` SvgIdUpdater.py` to see how the script works. The code should be pretty self-explanatory.
- Check the `/icons/elixir/elixir-test.svg` and `/icons/ruby/ruby-test.svg` again. You should see that the ids are properly replaced and both the "id" and "url()" will now point to the same new id.

Once everyone is happy with this, I will:
- Remove the test file
- Add code to loop through all `svg` and overwrite them.
- Make a new workflow that will do this for us from now on. I'm thinking it'll also be triggered via label like `bot:peek`. 

Note:
- Here's the doc for [`regexp.sub()`](https://docs.python.org/3/howto/regex.html#search-and-replace).
- Until we merge the new workflow in, please refrain from accepting new PR. This is so this new feature is apply from now on.

### CODE REVIEW

**Comment by Thomas-Boi** [comment]
Just tested the id-subbed files. Everything looks good:
image

I think I will sub the files now and push it as a new commit.

File: .github/scripts/build_assets/SvgIdUpdater.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+            new_content = search_regexp.sub(self.sub_id, file.read())
+
+            # write to a new file
+            # new_path = filepath.with_name(filepath.stem + "-written" + filepath.suffix)
+            # with open(new_path, "w") as new_file:
+            #     new_file.write(new_content)
+            #     print("Finished substituting. Writing to " + new_file.name)
+                
+            # overwrite same file
+            file.write(new_content)
```
Comment (maltejur) [inline .github/scripts/build_assets/SvgIdUpdater.py:53]: When testing out the script, it doesn't overwrite the svg file but just appends to it. 
```suggestion
            file.seek(0) 
            file.write(new_content)
            file.truncate()
```

**Review by maltejur — CHANGES_REQUESTED** [review]

**Comment by maltejur** [comment]
> I'm thinking it'll also be triggered via label like bot:peek

Why not just on every push to a pr?

File: .github/scripts/build_assets/SvgIdUpdater.py (hunk 1 above)
Comment (Thomas-Boi) [reply .github/scripts/build_assets/SvgIdUpdater.py]: Thanks for pointing this out. I also found this error yesterday when testing and forgot to push the change. The most recent commit should fix it (same code as yours)

**Comment by Thomas-Boi** [comment]
> Why not just on every push to a pr?

I thought about that as well and thought about what would happen when the file needs further changes? Most are minor but some of the bigger ones might make us rerun the script again. Triggering via label means we are sure the files are proper as svgs and as icons. It also prevents polluting the history with multiple commits after every new push with one minor changes.

**Comment by Thomas-Boi** [comment]
Here are the files that were updated:
```
aarch64-original.svg
aarch64-plain.svg
apache-original-wordmark.svg
apache-original.svg
babel-original.svg
babel-plain.svg
bash-original.svg
bash-plain.svg
behance-original-wordmark.svg
behance-original.svg
behance-plain-wordmark.svg
behance-plain.svg
cakephp-original-wordmark.svg
cakephp-original.svg
cakephp-plain-wordmark.svg
cakephp-plain.svg
ceylon-original-wordmark.svg
ceylon-original.svg
ceylon-plain-wordmark.svg
ceylon-plain.svg
chrome-original-wordmark.svg
chrome-original.svg
composer-line-wordmark.svg
composer-line.svg
composer-original.svg
couchdb-original-wordmark.svg
couchdb-original.svg
couchdb-plain-wordmark.svg
couchdb-plain.svg
cucumber-plain-wordmark.svg
cucumber-plain.svg
d3js-original.svg
d3js-plain.svg
devicon-original-wordmark.svg
devicon-original.svg
devicon-plain-wordmark.svg
devicon-plain.svg
digitalocean-original-wordmark.svg
digitalocean-original.svg
digitalocean-plain-wordmark.svg
digitalocean-plain.svg
dot-net-original-wordmark.svg
dot-net-original.svg
electron-original-wordmark.svg
electron-original.svg
elixir-original-wordmark.svg
elixir-original.svg
elixir-plain-wordmark.svg
elm-original-wordmark.svg
elm-original.svg
elm-plain-wordmark.svg
elm-plain.svg
ember-original-wordmark.svg
express-original-wordmark.svg
express-original.svg
facebook-original.svg
facebook-plain.svg
firefox-original-wordmark.svg
firefox-original.svg
flutter-original.svg
flutter-plain.svg
gatling-plain-wordmark.svg
gatling-plain.svg
gimp-original-wordmark.svg
gimp-original.svg
go-original.svg
google-original-wordmark.svg
google-original.svg
google-plain-wordmark.svg
google-plain.svg
gradle-plain-wordmark.svg
gradle-plain.svg
graphql-plain-wordmark.svg
graphql-plain.svg
handlebars-original-wordmark.svg
handlebars-original.svg
inkscape-original-wordmark.svg
inkscape-original.svg
intellij-original-wordmark.svg
intellij-original.svg
intellij-plain-wordmark.svg
intellij-plain.svg
ionic-original-wordmark.svg
ionic-original.svg
jasmine-plain-wordmark.svg
jasmine-plain.svg
jeet-original-wordmark.svg
jeet-original.svg
jenkins-line.svg
jenkins-original.svg
jenkins-plain.svg
jetbrains-original.svg
jupyter-original-wordmark.svg
jupyter-original.svg
jupyter-plain-wordmark.svg
jupyter-plain.svg
kotlin-original-wordmark.svg
kotlin-original.svg
kubernetes-plain.svg
linkedin-original-wordmark.svg
linkedin-original.svg
linkedin-plain-wordmark.svg
linkedin-plain.svg
linux-original.svg
lua-original-wordmark.svg
lua-original.svg
lua-plain-wordmark.svg
lua-plain.svg
matlab-original.svg
matlab-plain.svg
mocha-plain.svg
moodle-original-wordmark.svg
moodle-original.svg
npm-original-wordmark.svg
ocaml-original.svg
phoenix-original-wordmark.svg
phoenix-plain-wordmark.svg
php-original.svg
protractor-plain-wordmark.svg
protractor-plain.svg
pycharm-original-wordmark.svg
pycharm-original.svg
pycharm-plain-wordmark.svg
pycharm-plain.svg
python-original-wordmark.svg
python-original.svg
r-original.svg
rails-original-wordmark.svg
ruby-original-wordmark.svg
ruby-original.svg
rubymine-original-wordmark.svg
rubymine-original.svg
rubymine-plain-wordmark.svg
rubymine-plain.svg
safari-original-wordmark.svg
safari-original.svg
sequelize-original-wordmark.svg
sequelize-original.svg
sequelize-plain-wordmark.svg
sequelize-plain.svg
sketch-line-wordmark.svg
sketch-line.svg
sketch-original-wordmark.svg
sketch-original.svg
slack-original-wordmark.svg
slack-original.svg
slack-plain-wordmark.svg
slack-plain.svg
spring-original-wordmark.svg
spring-plain-wordmark.svg
sqlalchemy-original-wordmark.svg
sqlalchemy-original.svg
sqlalchemy-plain.svg
stylus-original.svg
swift-original-wordmark.svg
swift-original.svg
swift-plain-wordmark.svg
swift-plain.svg
thealgorithms-original-wordmark.svg
thealgorithms-original.svg
thealgorithms-plain-wordmark.svg
thealgorithms-plain.svg
twitter-original.svg
typescript-original.svg
typescript-plain.svg
uwsgi-original.svg
vagrant-original-wordmark.svg
vagrant-original.svg
vagrant-plain-wordmark.svg
vagrant-plain.svg
weblate-original-wordmark.svg
weblate-original.svg
weblate-plain-wordmark.svg
weblate-plain.svg
webpack-original-wordmark.svg
webpack-original.svg
webpack-plain-wordmark.svg
webpack-plain.svg
webstorm-original-wordmark.svg
webstorm-original.svg
webstorm-plain-wordmark.svg
webstorm-plain.svg
yarn-original-wordmark.svg
yarn-original.svg
yii-original-wordmark.svg
yii-original.svg
yii-plain-wordmark.svg
yunohost-original.svg
yunohost-plain.svg
```

Tested some of the svgs and they look good. Will go over each one in a bit.
image

**Comment by Thomas-Boi** [comment]
I will address the errors regarding our old icons. This is nothing big so all should be good.

**Comment by Thomas-Boi** [comment]
So my id subbing code kinda works but it misses some other svg references such as `href`. 

I also found out that [svgo](https://github.com/svg/svgo) also provides this prefix id functionality. They also provide svg optimization (which I suggested in this [discussion](https://github.com/devicons/devicon/discussions/593)). Thus, I'll pivot and use their code instead, which should keep everything consistent. 

New plan: changing the id subbing workflow proposed here into an svg optimization + id prefixing. I'll be closing this PR because the commit history has become a mess and open another one.

**Pull request closed without merging**